crates/uv-pep508/src/marker/parse.rs

```diff
@@ -101,14 +102,60 @@ pub(crate) fn parse_marker_value<T: Pep508Url>(
                 !char.is_whitespace() && !['>', '=', '<', '!', '~', ')'].contains(&char)
             });
             let key = cursor.slice(start, len);
-            MarkerValue::from_str(key).map_err(|_| Pep508Error {
-                message: Pep508ErrorSource::String(format!(
-                    "Expected a quoted string or a valid marker name, found `{key}`"
-                )),
-                start,
-                len,
-                input: cursor.to_string(),
-            })
+            MarkerValue::from_str(key)
+                .map_err(|_| Pep508Error {
+                    message: Pep508ErrorSource::String(format!(
+                        "Expected a quoted string or a valid marker name, found `{key}`"
+                    )),
+                    start,
+                    len,
+                    input: cursor.to_string(),
+                })
+                .inspect(|value| match value {
```

Comment:
nit: `?` to remove the inspect

Reply:
Is that better? Then I have to assign and return the value